tests/webdav-jest.ts

```diff
@@ -281,7 +282,7 @@ describe('Webdav integration', function testWebdavIntegration() {
       expect(await client.exists(subFolder3Path)).toBe(true);
 
       await client.remove(path);
-    });
+    }, 10000);
```

Comment:
Why did you introduce this wait?

Reply:
Because the tests were occasionally failing as they were taking longer than 5000ms.
The point of the test is verifying that nested folders work properly and testing just 2 folders isn't good enough.

Reply:
My reply this is the same as for our own code. Mock webdav client, don't use an actual Nextcloud container. We're not really testing whether Nextcloud works.

Unless this was an integration test, but then those tests should be clearly separate from any unit testing.

Reply:
The deal is, unit tests have very limited value here. What we really want to know is "does this work with real NC containers?", which is a question unit tests don't answer. We've seen multiple times in the existence od the library that unless we test this fully (integration), we have zero guarantee that the library works, because of quirks of the underlying Webdav library